### DISCUSSION TITLE
Quickly change CWD and back?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [LazyVim/LazyVim#2077] [Q&A] [answered]
Hello everyone, I work in a monorepo and jump around services a lot. It would be super helpful to somehow be able to hover over a folder and set it as the CWD, (for telescope search files and such). And then via another hotkey to go back to root as CWD. Any ideas how to achieve this? Is it even possible?

**Comment by dpetka2001** [comment]
If you have correctly configured your LSP to have the `root_dir` defined as the root of your monorepo, then doing `<leader>ff` will search files at the `root_dir` level downwards while `<leader>fF` will search files in your `cwd` when you're inside a service directory of your monorepo.

**Reply by Nikola-Milovic** [reply]
Yeah but if I am inside some nested folders in service A, then CWD is not the root of service A but the subfolder. Or am I missing something?

**Reply by dpetka2001** [reply, marked as answer]
Yes `cwd` is always the current directory where you're launching Neovim. I don't understand your workflow. Usually, I open Neovim at the root of the folder and then use Telescope or fzf-lua to fuzzy find the files I want. If I want some specific files in different folders which I go back and forth I use Harpoon to pin them in a toggleable pop-up window for quick access. You could also make a new keymap for Telescope find_files to look in the directory of the file that you have open in your current buffer. Something like 
```lua
{
  "nvim-telescope/telescope.nvim",
  keys = {
    { "<leader>fz", "<cmd>Telescope find_files cwd=%:p:h<cr>", desc = "Find files (dir of current file)" },
  },
},
```
should work I believe. This keymap will always search in the directory of the file in your current buffer. Then you can use `<leader>ff` for regular `root_dir` search.
